### PR TITLE
Added permissions to watch IR report

### PR DESCRIPTION
**Pull request by aklochkova**
Fixes https://github.com/OHDSI/WebAPI/issues/985

### CODE REVIEW

File: src/main/resources/db/migration/sqlserver/V2.7.0.20190328172203__added-ir-report-perms-for-sources.sql [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+INSERT INTO #Seq_values
+SELECT 0, REPLACE(new_perms.val, '%s', REPLACE(REPLACE(value, 'source:', ''), ':access', ''))
+FROM ${ohdsiSchema}.sec_permission sp
+       JOIN ${ohdsiSchema}.sec_role_permission srp on sp.id = srp.permission_id
+       CROSS JOIN (
+  SELECT 'ir:*:report:%s:get' val
+) new_perms
+WHERE sp.value LIKE 'source:%:access';
+
+UPDATE #Seq_values
```
Comment (wivern) [inline src/main/resources/db/migration/sqlserver/V2.7.0.20190328172203__added-ir-report-perms-for-sources.sql]: why we need this update but not to use sequence directly in the previous insert?

File: src/main/resources/db/migration/sqlserver/V2.7.0.20190328172203__added-ir-report-perms-for-sources.sql [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  WHERE sp.value LIKE 'source:%:access'
+)
+INSERT
+INTO ${ohdsiSchema}.sec_permission (id, value, description, role_id_tmp)
+SELECT perms_for_all_sources.id,
+       perms_for_all_sources.value,
+       perms_for_all_sources.description,
+       perms_for_all_sources.role_id
+FROM perms_for_all_sources EXCEPT
+SELECT perms_for_all_sources.id,
```
Comment (wivern) [inline src/main/resources/db/migration/sqlserver/V2.7.0.20190328172203__added-ir-report-perms-for-sources.sql]: why not to use `NOT EXISTS` which if free of `EXCEPT` side-effects?

File: src/main/resources/db/migration/postgresql/V2.7.0.20190328172203__added-ir-report-perms-for-sources.sql [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  WHERE sp.value LIKE 'source:%:access'
+)
+INSERT
+INTO ${ohdsiSchema}.sec_permission (id, value, description, role_id_tmp)
+SELECT perms_for_all_sources.id,
+       perms_for_all_sources.value,
+       perms_for_all_sources.description,
+       perms_for_all_sources.role_id
+FROM perms_for_all_sources EXCEPT
+SELECT perms_for_all_sources.id,
```
Comment (wivern) [inline src/main/resources/db/migration/postgresql/V2.7.0.20190328172203__added-ir-report-perms-for-sources.sql]: I guess this expression cuts out **all permission records** if at least one role has comparing permission

File: src/main/resources/db/migration/oracle/V2.7.0.20190328172203__added-ir-report-perms-for-sources.sql [hunk 4]
```diff
@@ -0,0 +1,44 @@
+ALTER TABLE ${ohdsiSchema}.sec_permission
+  ADD role_id_tmp INTEGER;
+
+CREATE OR REPLACE FUNCTION ${ohdsiSchema}.get_next_sequence_val RETURN NUMBER IS
```
Comment (wivern) [inline src/main/resources/db/migration/oracle/V2.7.0.20190328172203__added-ir-report-perms-for-sources.sql]: why we need this function?
and where is it dropped?

File: src/main/resources/db/migration/oracle/V2.7.0.20190328172203__added-ir-report-perms-for-sources.sql [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+         CROSS JOIN (
+    SELECT 'ir:*:report:%s:get' val, 'Get IR generation report with SourceKey = %s' descr FROM DUAL) new_perms
+  WHERE sp.value LIKE 'source:%:access'
+)
+SELECT perms_for_all_sources.id,
+       perms_for_all_sources.value,
+       perms_for_all_sources.description,
+       perms_for_all_sources.role_id
+FROM perms_for_all_sources
+MINUS
```
Comment (wivern) [inline src/main/resources/db/migration/oracle/V2.7.0.20190328172203__added-ir-report-perms-for-sources.sql]: same as for PgSQL

**Comment by chrisknoll** [comment]
@wivern , please flush the toilet.

**Pull request merged**